atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/LightweightOppToken.java

```diff
@@ -34,6 +37,35 @@ public LightweightOppToken(byte[] bytes) {
         this.bytes = bytes;
     }
 
+    public static LightweightOppToken fromHex(String token) {
+        // OPP tokens should be lowercase already, use upper if needed, and convert mixed to uppercase if needed
+        byte[] bytes = isAllLowercaseOrDigits(token)
+                ? BaseEncoding.base16().lowerCase().decode(token)
+                : isAllUppercaseOrDigits(token)
+                        ? BaseEncoding.base16().decode(token)
+                        : BaseEncoding.base16().decode(token.toUpperCase(Locale.ROOT));
+        return new LightweightOppToken(bytes);
+    }
+
+    @VisibleForTesting
+    static boolean isAllLowercaseOrDigits(String token) {
```

Comment:
Could we take this + the other methods `isAllUppercaseOrDigits` and `allMatch` and throw these into a separate utility class? Maybe `PerformantStringUtils` or something of the sort?

Reply:
Plus then we can have separate easy tests for them :)

Reply:
I can make the private functions visible for tests and add test coverage for them, but these utilities aren't and really shouldn't be visible for use elsewhere, and are specific for use converting cassandra order preserving partition tokens.

Reply:
Separate from this, but we really need to update AtlasDB to JUnit 5 so that we can leverage better parameterized tests.